packages/components/src/focal-point-picker/index.native.js

```diff
@@ -0,0 +1,264 @@
+			}
+		} );
+	}, [] );
+
+	const pan = useRef( new Animated.ValueXY() ).current;
+
+	// Set initial cursor poition
+	useEffect( () => {
+		if ( containerSize ) {
+			pan.setValue( {
+				x: focalPoint.x * containerSize.width,
+				y: focalPoint.y * containerSize.height,
+			} );
+		}
+	}, [ focalPoint, containerSize ] );
+
+	const panResponder = useMemo(
+		() =>
+			PanResponder.create( {
+				onStartShouldSetPanResponder: () => true,
+				onStartShouldSetPanResponderCapture: () => true,
+				onMoveShouldSetPanResponder: () => true,
+				onMoveShouldSetPanResponderCapture: () => true,
+
+				onPanResponderGrant: ( event ) => {
+					shouldEnableBottomSheetScroll( false );
+					const { locationX: x, locationY: y } = event.nativeEvent;
+					pan.setValue( { x, y } ); // Set cursor to tap location
+					pan.extractOffset(); // Set offset to current value
+				},
+				// Move cursor to match delta drag
+				onPanResponderMove: Animated.event( [
+					null,
+					{ dx: pan.x, dy: pan.y },
+				] ),
+				onPanResponderRelease: ( event ) => {
+					shouldEnableBottomSheetScroll( true );
+					pan.flattenOffset(); // Flatten offset into value
+					const { locationX: x, locationY: y } = event.nativeEvent;
+					setPosition( {
+						x: x / containerSize?.width,
+						y: y / containerSize?.height,
+					} );
+					// Slider (child of RangeCell) is uncontrolled, so we must increment a
+					// key to re-mount and sync the pan gesture values to the sliders
+					// https://git.io/JTe4A
+					setSliderKey( ( prevState ) => prevState + 1 );
+				},
+			} ),
+		[ containerSize ]
+	);
+
+	function setPosition( { x, y } ) {
+		onChange( ( prevState ) => ( {
+			...prevState,
+			...( x ? { x } : {} ),
+			...( y ? { y } : {} ),
+		} ) );
+	}
+
+	const backgroundColor = getStylesFromColorScheme(
+		styles.backgroundSolid,
+		styles.backgroundSolidDark
+	);
+	const imagePreviewStyles = [
+		displayPlaceholder && styles.mediaPlaceholder,
+	];
+	const videoPreviewStyles = [
+		{
+			aspectRatio:
+				videoNaturalSize &&
+				videoNaturalSize.width / videoNaturalSize.height,
+			height: '100%',
+		},
+		! displayPlaceholder && styles.video,
+		displayPlaceholder && styles.mediaPlaceholder,
+	];
+	const focalPointGroupStyles = [
+		styles.focalPointGroup,
+		{
+			transform: [
+				{
+					translateX: pan.x.interpolate( {
+						inputRange: [ 0, containerSize?.width || 0 ],
+						outputRange: [ 0, containerSize?.width || 0 ],
+						extrapolate: 'clamp',
+					} ),
+				},
+				{
+					translateY: pan.y.interpolate( {
+						inputRange: [ 0, containerSize?.height || 0 ],
+						outputRange: [ 0, containerSize?.height || 0 ],
+						extrapolate: 'clamp',
+					} ),
+				},
+			],
+		},
+	];
+
+	const FOCAL_POINT_SIZE = 50;
+	const focalPointStyles = [
+		styles.focalPoint,
+		{
+			height: FOCAL_POINT_SIZE,
+			marginLeft: -( FOCAL_POINT_SIZE / 2 ),
+			marginTop: -( FOCAL_POINT_SIZE / 2 ),
+			width: FOCAL_POINT_SIZE,
+		},
+	];
+
+	return (
```

Comment:
I see there are some arrow functions within the return that will create new references every time there's a re-render. What do you think if we move them outside of it?

Reply:
@geriux happy to relocate them to avoid additional function expressions.

In general, I tend to not worry about inlined anonymous functions within render. This is primarily based upon opinions and guidance I have read from the React core team and prominent React community members. That said, I completely understand it does have a cost, and I'm happy to refactor them away.

- [Comment from Dan Abramov on a related Twitter thread.](https://twitter.com/dan_abramov/status/1069246963898568706)
- [React Hooks documentation on the impact of functions within render.](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render)
- [Blog post by Ryan Florence on inline functions.](https://reacttraining.com/blog/react-inline-functions-and-performance/)

I will say that the links I provide and most of the conversation I have observed have been in the context of web development, but my take-away was that it would still be applicable to React Native and the JavaScript runtime used there.

What are your thoughts on my understanding of the cost of inline functions?

Reply:
For mobile, it is a bit different yes and we do have to take those costs into account, especially for lower-end devices. Here's a quick [post](https://medium.com/@adamjacobb/react-native-performance-arrow-functions-binding-3f09cbd57545) I found.